### ISSUE TITLE
Create RPM package for Fedora/CentOS

### ISSUE THREAD

**Issue by francescmm** [francescmm/GitQlient#74]
In the same way that #73, it would be nice to have the RPM package for Fedora/CentOS.

**Comment by hegjon** [comment]
I am a Fedora developer and maintain several packages in the official Fedora repo. What is your goal? Do you want to host it yourself or do you want GitQlient to be distributed by Fedora and other RPM distros?

**Comment by hegjon** [comment]
I made some efforts to build on https://copr.fedorainfracloud.org/

But COPR seems to not support git sub-modules out of the box.

Error log:
https://download.copr.fedorainfracloud.org/results/jonny/Qt/fedora-32-x86_64/01684654-gitqlient/build.log.gz

The source tarball included in the releases do not include the QLogger either, so changing the RPM spec to build from source tarball will also fail.

**Comment by francescmm** [comment]
Hi @hegjon!

So ideally I'd like to have it distributed by Fedora and other RPM distros but if that's impossible or adds a huge workload I can try a different approach.

About QLogger, I didn't thought on that, so I'll probably move it into the code and remove the submodule. My bad for not checking it :)

The problem is that I don't have any experience on it, so I'd need as much help as possible :smile:

**Comment by hegjon** [comment]
If you produce a tarball with the QLogger included then it should be easy to build

**Comment by hegjon** [comment]
As a side note, I am learning Qt and QML. I have been using the built-in logging framework in Qt, what are the limitations of the built-in logging and how does QLogger solve them? There is very limited information in the QLogger repo.

**Comment by francescmm** [comment]
I can generate the tarball for a commit, but I don't know what I have to do in COPR about it. I couldn't find a stpe-by-step manual that guides me through the process. What would I need to do?

About QLogger the main difference is the support for writing into files and truncate them when they are too big. It also controls the configuration of _how_ to log (per file, per module, etc). Finally it adds multithreading support as well as control over the level in a more transparent way. The logging system of Qt is quite _debug oriented_ not that suitable for deployment. I will add more documentation into it, at lest about the benefits.

**Comment by hegjon** [comment]
> I can generate the tarball for a commit, but I don't know what I have to do in COPR about it. I couldn't find a stpe-by-step manual that guides me through the process. What would I need to do?

COPR support building directly from Git, I tried with my fork https://github.com/hegjon/GitQlient/tree/copr/contrib/rpm. The build can be triggered by per commit or for instance manually after you have done a release. I can help you with making the RPM spec if needed. It is also possible to let COPR build for CentOS, openSUSE and Mageia.

To have GitQlient included in Fedora, then I only need a source tarball that can be built without internet access (no git submodule calls)

**Comment by hegjon** [comment]
make install is not setup properly, it is not possible to override the destination location
```
jonny@jonny-thinkpad /tmp/rpkg/gitqlient-11-n6ueup11/gitqlient-0.0.git.1509.c27f631.wtree.0rhsps{12.16.1} $ /usr/bin/make install DESTDIR=/tmp/rpkg/gitqlient-11-n6ueup11/gitqlient-1.2.0-0.0.git.1509.c27f631.wtree.0rhsps.fc32.x86_64 'INSTALL=/usr/bin/install -p'
/usr/lib64/qt5/bin/qmake -install qinstall -exe GitQlient /home/jonny/GitQlient/bin/GitQlient
```

**Comment by francescmm** [comment]
For projects that run witn QMake as it is GitQlient you need to set the destination during qmake execution, so [in your case](https://github.com/hegjon/GitQlient/blob/copr/contrib/rpm/gitqlient.spec#L30) would be adding DESTDIR to those lines.

**Comment by francescmm** [comment]
It would be nice if you could PR your changes or drive me through the things I would need to do to provide the RPMs periodically. Do you think that's possible?

**Comment by hegjon** [comment]
The latest builds for several RPM distroes https://copr.fedorainfracloud.org/coprs/jonny/Qt/build/1688015/

The settings used:
```
Package: gitqlient
Version: 1.2.0-0.0.git.1522.e55fed9.fc32
Source Type: Build from an SCM repository
SCM type: git
Clone URL: https://github.com/hegjon/GitQlient.git
Committish: copr
Path to .spec file: contrib/rpm/gitqlient.spec
Build SRPM with: rpkg
```

**Comment by francescmm** [comment]
OMG this looks amazing! I'm gonna try it tomorrow!

**Comment by francescmm** [comment]
Do you know what I would need to do (or who to ask) to include the packages in the official repos?

**Comment by hegjon** [comment]
> OMG this looks amazing! I'm gonna try it tomorrow!

Thanks, great to hear

**Comment by hegjon** [comment]
> Do you know what I would need to do (or who to ask) to include the packages in the official repos?

It takes a bit of time to become approved as a packager for Fedora. I am not sure how openSUSE and Mageia maintains it's official packages. What distro are you running?

I can add it and maintain it in the official Fedora repo, since I am already maintaining other packages in Fedora [1]. There are some changes that should be done first, if you are interested then I can make a new issue that lists those.

[1] https://src.fedoraproject.org/user/jonny/projects

**Comment by francescmm** [comment]
> It takes a bit of time to become approved as a packager for Fedora. I am not sure how openSUSE and Mageia maintains it's official packages. What distro are you running?
> 
> I can add it and maintain it in the official Fedora repo, since I am already maintaining other packages in Fedora [1]. There are some changes that should be done first, if you are interested then I can make a new issue that lists those.
> 
> [1] https://src.fedoraproject.org/user/jonny/projects

I'm actually running Ubuntu (I was a Mandrake/Mandriva user in the old days but... we now how that ended).

I would be happy if you want to maintain the package! I really think it would add an extra thing to take care and I don't think I can afford it.

So, please open another issue with the required changes. I'm happy to adapt whatever the project needs for it.

**Comment by francescmm** [comment]
I'll consider this done and we can follow up the integration of the RPMs packages in another issue.
